### PR TITLE
Products page with filter functionality

### PR DESCRIPTION
**Pull request by aayushjain-code**

### CODE REVIEW

**Comment by aayushjain-code** [comment]
https://subtle-cat-b2ce8f.netlify.app/products

File: src/Components/Products/FilterContainer.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+					<option value="Above 100 Below 200">Above 100 Below 200</option>
+					<option value="Above 200 Below 300">Above 200 Below 300</option>
+				</select>
+			</span>
+
+			<span className="filter-sub-heading">
+				<label>
+					<input type="radio" name="group1"
+						onChange={() => filterDispatch({ type: 'SORT_BY_PRICE', payload: 'lowToHigh' })}
+						checked={filterState.isSort === 'lowToHigh' ? true : false}
```
Comment (govindappaarun) [inline src/Components/Products/FilterContainer.js:24]: suggestion you don't need a ternary, simplify

File: src/Components/Products/FilterContainer.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+					/> <span>Low to high</span>
+				</label>
+				<br />
+				<label>
+					<input type="radio" name="group1"
+						onChange={() => filterDispatch({ type: 'SORT_BY_PRICE', payload: 'highToLow' })}
+						checked={filterState.isSort === 'highToLow' ? true : false}
```
Comment (govindappaarun) [inline src/Components/Products/FilterContainer.js:31]: same comment here

File: src/Components/Products/FilterContainer.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+						<label>
+							<input type="radio" name="group2" onChange={() => filterDispatch({ type: 'FILTER_BY_RATING', payload: '2StarAndAbove' })}
+								checked={filterState.byRating === '2StarAndAbove' ? true : false} />
+							<p style={{ marginLeft: '2rem', marginTop: '-2.3rem' }}>2 Star and Above</p>
+						</label>
+					</span>
+					<span>
+						<label>
+							<input type="radio" name="group2" onChange={() => filterDispatch({ type: 'FILTER_BY_RATING', payload: '1StarAndAbove' })}
+								checked={filterState.byRating === '1StarAndAbove' ? true : false} />
```
Comment (govindappaarun) [inline src/Components/Products/FilterContainer.js:101]: the equality gives boolean value, so no need of ternary, and you need to update this in many conditions

Comment (govindappaarun) [reply src/Components/Products/FilterContainer.js:101]: suggestion: use code format tools like prettier, hard to read your code on screen

File: src/Components/Products/Products.js [hunk 4]
```diff
@@ -1,9 +1,47 @@
-import React from 'react'
-import "./products.css"
+// import { useState, useEffect } from 'react'
+import ProductCard from './ProductCard'
+import axios from 'axios';
+import "./products.css";
+import { useFilter } from '../../context/context';
+import FilterContainer from './FilterContainer';
+import { products } from "../../backend/db/products"
```
Comment (govindappaarun) [inline src/Components/Products/Products.js:7]: you should not import products from backend
and I see few unused imports

File: src/Components/Products/updatedProducts.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { useFilter } from "../../context/context";
+
+
+export const FilterFunction = (products) => {
```
Comment (govindappaarun) [inline src/Components/Products/updatedProducts.js:4]: suggestion: functions should be named with camelCase

File: src/context/context.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const Context = createContext();
+
+const Provider = ({ children }) => {
+
+	const [filterState, filterDispatch] = useReducer(filterReducer, {
+		excludeNotAvailable: false,
+		offerItems: false,
+		byRating: "",
+		isSort: "",
```
Comment (govindappaarun) [inline src/context/context.js:12]: you can put this as a constant and import in reducer also to avoid code duplication

**Review by govindappaarun — COMMENTED** [review]
I see functionality wise looks fine, Good thing I see is modular. 
Improve on the coding standards and try for making  code more reusable

File: src/Components/Products/updatedProducts.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	if (filterState.byCategory) {
+		if (filterState.byCategory === 'All') {
+			updatedProductList = products
+		} else {
+			updatedProductList = updatedProductList.filter((item) => item.categoryName === filterState.byCategory)
+			console.log(updatedProductList)
+		}
+	}
+
+	if (filterState.byPrice) {
```
Comment (MuskanChhatrasal) [inline src/Components/Products/updatedProducts.js:27]: Start using prettier for code inclination as it is hard to read. 
UI is good, functionality is working properly. Good to merge

**Review by MuskanChhatrasal — APPROVED** [review]

**Pull request merged**